src/types/types.ts

```diff
@@ -75,18 +76,18 @@ export interface Cell {
   y: number; // y position of cell in dataset
   pageX: number; // X position of cell in page
   pageY: number; // Y position of cell in page
+  mainAxisPageCoord: number; // Either equal pageX or pageY depending on if a cell is in the left or top grid
```

Comment:
Borrowed the "main axis" part from css flexbox. Maybe bit too contrived name, but very useful.